test/uk/gov/hmrc/securemessage/controllers/SecureMessageControllerSpec.scala

```diff
@@ -76,6 +78,35 @@ class SecureMessageControllerSpec extends PlaySpec with ScalaFutures with Mockit
       private val response = controller.createConversation("cdcm", "123")(fakeRequest)
       status(response) mustBe BAD_REQUEST
     }
+
+    "return CONFLICT (409) when there is an error storing the conversation" in new CreateConversationTestCase(
+      requestBody = Resources.readJson("model/api/create-conversation-full.json")) {
+      when(mockSecureMessageService.createConversation(any[Conversation])(any[HeaderCarrier], any[ExecutionContext]))
+        .thenReturn(Future(Left(StoreException(CONFLICT, "conflict error", None))))
+      private val response = controller.createConversation("cdcm", "123")(fakeRequest)
+      status(response) mustBe CONFLICT
+      contentAsString(response) mustBe "conflict error"
+    }
+
+    "return CREATED (201) when there is an error sending the email" in new CreateConversationTestCase(
+      requestBody = Resources.readJson("model/api/create-conversation-full.json")) {
+      when(mockSecureMessageService.createConversation(any[Conversation])(any[HeaderCarrier], any[ExecutionContext]))
+        .thenReturn(Future(Left(EmailException(CREATED, "email error"))))
```

Comment:
this makes me think we should remove code from email exception and assign an HTTP code at the controller level when we handle exceptions.

Reply:
I have removed the code from all classes